src/MagnumPlugins/BasisImageConverter/Test/BasisImageConverterTest.cpp

```diff
@@ -1425,6 +1429,61 @@ void BasisImageConverterTest::swizzle() {
         TestSuite::Compare::around(Vector4i{2}));
 }
 
+void BasisImageConverterTest::openCL() {
+    if(_manager.loadState("PngImporter") == PluginManager::LoadState::NotFound)
+        CORRADE_SKIP("PngImporter plugin not found, cannot test contents");
+
+    Containers::Pointer<AbstractImporter> pngImporter = _manager.instantiate("PngImporter");
+    CORRADE_VERIFY(pngImporter->openFile(Utility::Path::join(BASISIMPORTER_TEST_DIR, "rgba-31x13.png")));
+    Containers::Optional<Trade::ImageData2D> original = pngImporter->image2D(0);
+    CORRADE_VERIFY(original);
+
+    Containers::Pointer<AbstractImageConverter> converter = _converterManager.instantiate("BasisImageConverter");
+    /* Default is off */
+    CORRADE_COMPARE(converter->configuration().value<bool>("use_opencl"), false);
+    converter->configuration().setValue("use_opencl", true);
+
+    std::ostringstream out;
+    Containers::Optional<Containers::Array<char>> compressedData;
+    {
+        Warning redirectWarning{&out};
+        compressedData = converter->convertToData(*original);
+    }
+
+    /* If built without OpenCL, converting falls back to CPU and still succeeds */
+    #ifndef OpenCL_FOUND
```

Comment:
I guess this needs another branch with `#if BASISU_VERSION < 115`, with `CORRADE_INFO("OpenCL not used in version 1.15 yet.");`.

That's what is making the MSVC 2017 build to fail.

Reply:
I went the "easy" route and just added `BasisUniversalEncoder_INCLUDE_DIR` to the test so it has access to the version define in one of the Basis headers. Of course this broke on GCC < 5 because that needs the `__is_trivially_copyable=__has_trivial_copy` fix on the actual Encoder target. I fixed that by copying the INTERFACE_COMPILE_DEFINITIONS from the Encoder, but I'm wondering if this is already too ugly.

Is just linking to the Encoder target from the test the better way? I'm just a bit afraid of weird side effects this might have on coverage etc.

Reply:
Linking `Encoder` target could do quite a mess on static builds, suddenly everything being twice in there etc. 

Copying the compile definitions is fine, it's quite clean actually :)